viewer/arvviewer.c

```diff
@@ -241,8 +241,30 @@ arv_viewer_value_from_log (double value, double min, double max)
 	return pow (10.0, (value * (log10 (max) - log10 (min)) + log10 (min)));
 }
 
+struct free_arv_buffer_cb_param
+{
+	GWeakRef stream;
+	GWeakRef arv_buffer;
+};
+
+static void
+free_arv_buffer_cb (void* param)
+{
+	struct free_arv_buffer_cb_param* cb_param = (struct free_arv_buffer_cb_param*)param;
+
+	ArvStream* stream = g_weak_ref_get (&cb_param->stream);
```

Comment:
I did not know this API, thanks !
 Reading the documentation, g_weak_ref converts the weak reference to a normal reference. That probably means there is a reference leak in this function. You should call g_object_unref line 260.